### ISSUE TITLE
Core bundles do not start on Karaf 3.0.3 (OpenDayLight)

### ISSUE THREAD

**Issue by TimVinkemeier** [fp7-netide/Engine#23]
As @peppepetra  pointed out, the core currently does not run on Karaf 3.0.3. I will copy the email conversation here, so that everyone can track it:

@peppepetra:
I am trying to run Core bundles inside Karaf 3.0.3 that is the version used by Opendaylight Openflowplugin, that is the environment where the odl-shim should run, but there are a lot of dependencies not satisfied as core bundles are built using Karaf 4.0.0. 
Is it possible to downgrade the dependency to 3.0.3 or using 4.0.0 is a requirement for the core?

@TimVinkemeier:
I have no idea, when I started development, I just used the newest version that is available. Which are the dependencies that are missing in Karaf 3.0.3?

@peppepetra:
Unfortunately as the missing constraints are for package used at runtime, I can’t know what are all the missing packages.
The two first missing requirements are:
-   org.apache.karaf.config.core version [4.0.0, 5.0.0)                 karaf 3.0.3 exports version 3.0.3
-   org.osgi.framework [1.8.0,2.0.0)                                       Karaf 3.0.3 exports version 1.7.0

@TimVinkemeier:
org.apache.karaf.config.core is a dependency that we can not easily change, since it is integral to the management module.
Regarding org.osgi.framework dependency, we could just try to include 1.7.0 in the range and see if that works.

**Comment by TimVinkemeier** [comment]
Maybe you could try to change the org.osgi.framework dependencies and run the core modules without the management bundle?

**Comment by peppepetra** [comment]
I changed the org.osgi.framework version and I run the core without the management and it seems to work. At least bundles are activated with no error.

**Comment by TimVinkemeier** [comment]
Okay, I thought so, at least then it is easy to use them.
That still leaves us with the Management Bundle, though...
Do we have any Information on which Version of Karaf will be used for the next ODL release?
Otherwise we would have to find out how the mechanism works in 3.0.3 or if it is not present at all...

**Comment by TimVinkemeier** [comment]
I just downgraded the core to run on Karaf 3.0.3.
Interestingly, I did not have any issue with the org.osgi.framework dependency, since none of the bundles uses this (anymore, as it seems) :smiley:.
I only had to change the org.apache.karaf.config.core dependency to 3.0.3 and the features.xml schema version to 1.2.0 (Karaf 3.0.4 would also support the new 1.3.0 schema, but 3.0.3 does not).

In conclusion: Since bf94f94, the core is able to run on Karaf 3.0.3.
